core/dictgen/src/rootcling_impl.cxx

```diff
@@ -3218,7 +3218,7 @@ class tempFileNamesCatalog {
    tempFileNamesCatalog(): m_size(0), m_emptyString("") {};
 
    std::string getTmpFileName(const std::string &filename) {
-      return filename + "_tmp";
+      return filename + "_tmp_" + std::to_string(getpid());
```

Comment:
I find this problematic - this will mean we'll never fix the underlying build system issue that triggers concurrent invocations on the same dictionary. Could you add a note to the relevant Jira ticket https://sft.its.cern.ch/jira/browse/ROOT-8244 that this commit should be reverted once the issue is fixed?